### ISSUE TITLE
Revisit the resource caching requirement

### ISSUE THREAD

**Issue by sabbyanandan** [spring-cloud/spring-cloud-deployer#235]
As a user, while deploying a stream using SCDF to K8S, I see the logs regarding caching the docker images, which seems irrelevant for non-maven artifacts. Though we [log the inapplicability](https://github.com/spring-cloud/spring-cloud-deployer/blob/master/spring-cloud-deployer-resource-support/src/main/java/org/springframework/cloud/deployer/resource/support/DelegatingResourceLoader.java#L128), it seems like we are still [creating a cache directory](https://github.com/spring-cloud/spring-cloud-deployer/blob/master/spring-cloud-deployer-resource-support/src/main/java/org/springframework/cloud/deployer/resource/support/DelegatingResourceLoader.java#L121), and that is not required. Perhaps we can remove the directory after logging the `WARN` message?

Logs:
```
2017-10-23 15:29:42.866  INFO 80812 --- [nio-7577-exec-3] o.s.c.d.r.s.DelegatingResourceLoader     : Caching resource 'docker:springcloudstream/log-sink-rabbit:1.2.0.RELEASE' as file /var/folders/c3/ctx7_rns6x30tq7rb76wzqwr0000gs/T/deployer-resource-cache3504604345247190928/docker-293360093b9ac0e97da54d2c95043fa08a8d6e2d
2017-10-23 15:29:42.866  WARN 80812 --- [nio-7577-exec-3] o.s.c.d.r.s.DelegatingResourceLoader     : Unable to cache file since getInputStream() is not supported for resource: Docker Resource [docker:springcloudstream/log-sink-rabbit:1.2.0.RELEASE]
2017-10-23 15:29:43.037  INFO 80812 --- [nio-7577-exec-3] o.s.c.d.s.k.DefaultContainerFactory      : Using Docker image: springcloudstream/log-sink-rabbit:1.2.0.RELEASE
2017-10-23 15:29:43.037  INFO 80812 --- [nio-7577-exec-3] o.s.c.d.s.k.DefaultContainerFactory      : Using Docker entry point style: exec
2017-10-23 15:29:44.277  INFO 80812 --- [nio-7577-exec-3] o.s.c.d.r.s.DelegatingResourceLoader     : Caching resource 'docker:springcloudstream/time-source-rabbit:1.2.0.RELEASE' as file /var/folders/c3/ctx7_rns6x30tq7rb76wzqwr0000gs/T/deployer-resource-cache3504604345247190928/docker-2b9b611e16fd13c9487f3c23d9e33928663c84f7
2017-10-23 15:29:44.277  WARN 80812 --- [nio-7577-exec-3] o.s.c.d.r.s.DelegatingResourceLoader     : Unable to cache file since getInputStream() is not supported for resource: Docker Resource [docker:springcloudstream/time-source-rabbit:1.2.0.RELEASE]
```

**Comment by sabbyanandan** [comment]
Depending on what we decide here, #193 may not be required. It might still be useful to have a feature enable/disable toggle.

**Comment by sabbyanandan** [comment]
Based on the recent discussion, we will refactor the caching routine with the following approach.

1) `DelegatingResourceLoader` implementation will default to [LRU design implemented for the cf-server](https://github.com/spring-cloud/spring-cloud-dataflow-server-cloudfoundry/blob/master/spring-cloud-dataflow-server-cloudfoundry-autoconfig/src/main/java/org/springframework/cloud/dataflow/server/cloudfoundry/resource/LRUCleaningResourceLoader.java) (we will move it to deployer)

2) Take into account `scheme` in the `getResource()` implementation

3) Don't do anything for `docker` artifacts (they don't need to be looked-up or cached; they are passed as strings to the deployer)

4) Following platform-specific resolution model needs to be verified

```
local -> download for maven and http; docker - no need to download
cf -> download for maven and http; docker - no need to download
k8s -> do not download anything
```

5) Verify that the metadata resources follow the same model as discussed above

**Comment by sabbyanandan** [comment]
Bullets 2, 3, 4, and 5 are addressed via: spring-cloud/spring-cloud-deployer#239. The only item that remains is the LRU port from the cf-server.
